libraries/resource_providers/api300/c7000/scope_provider.rb

```diff
@@ -0,0 +1,56 @@
+# (c) Copyright 2017 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+require_relative '../../../resource_provider'
+
+module OneviewCookbook
+  module API300
+    module C7000
+      # Scope API300 C7000 provider methods
+      class ScopeProvider < ResourceProvider
+        # Adds or removes resources to a specified scope
+        # @note Calls the parse_resource_values method to retrieve the resources
+        # @return [<OneviewSDK::<APIVER>::<VARIANT>::Scope>] with the updated resources
+        def change_resource_assignments
+          add_or_remove = @context.add.any? || @context.remove.any?
+          raise "Unspecified properties: 'add' and 'remove'. Please set at least one before attempting this action." unless add_or_remove
+          @item.retrieve!
+          values_to_add = parse_resource_values
+          values_to_remove = parse_resource_values(:remove)
+          @item.change_resource_assignments(add_resources: values_to_add, remove_resources: values_to_remove)
+        end
+
+        # Retrieves the resources listed on the add or remove fields and creates an array with their instances
+        # @return [Array] containing the instances of all resources to be added or removed
+        def parse_resource_values(op = :add)
```

Comment:
It verifies the combinations of `<resource_type>: [<resource_names>]` that were listed in the `add` and `remove` fields, tries to retrieve each resource (failing if any does not exist), and verifies if the resource is already assigned to the scope, so that it does not try to add or remove a resource needlessly.
`parse` was the best I could come up with, but I'm entirely up to suggestions.